src/gui/guiFormSpecMenu.cpp

```diff
@@ -1326,6 +1327,50 @@ void GUIFormSpecMenu::parseField(parserData* data, const std::string &element,
 	errorstream<< "Invalid field element(" << parts.size() << "): '" << element << "'"  << std::endl;
 }
 
+void GUIFormSpecMenu::parseHyperText(parserData* data, const std::string &element)
+{
+	std::vector<std::string> parts = split(element,';');
+	if (parts.size() == 4)
+	{
+		std::vector<std::string> v_pos = split(parts[0],',');
+		std::vector<std::string> v_geom = split(parts[1],',');
+		std::string name = parts[2];
+		std::string text = parts[3];
+
+		MY_CHECKPOS("text",0);
+		MY_CHECKGEOM("text",1);
+
+		// [ Taken from textarea
+		v2f32 pos = pos_offset * spacing;
+		pos.X += stof(v_pos[0]) * (float) spacing.X;
+		pos.Y += stof(v_pos[1]) * (float) spacing.Y + m_btn_height;
+
+		v2s32 geom;
+		geom.X = (stof(v_geom[0]) * (float)spacing.X) - (spacing.X-imgsize.X);
+		geom.Y = (stof(v_geom[1]) * (float)imgsize.Y) - (spacing.Y-imgsize.Y);
+
+		core::rect<s32> rect = core::rect<s32>(pos.X, pos.Y, pos.X+geom.X, pos.Y+geom.Y);
+		// ]
+		if(m_form_src)
+			text = m_form_src->resolveText(text);
+
+		FieldSpec spec(
+			name,
+			utf8_to_wide(unescape_string(text)),
+			L"",
+			258+m_fields.size()
+		);
+
+		spec.ftype = f_Unknown;
+		new GUIHyperText(spec.flabel.c_str(), Environment, this, spec.fid, rect, m_client, m_tsrc);
```

Comment:
Where is it destructed? Usually you have to reduce the reference counter by 1 so that Irrlicht can destroy the element as soon it's not displayed any more.

Reply:
Destructing it in the dtor causes a double-free. Looks like it's dropped somewhere else (possibly in `removeChildren`?)